src/lib/FreeC/Frontend/Haskell/Simplifier.hs

```diff
@@ -328,8 +328,12 @@ simplifyDecl decl@(HSE.GDataDecl _ _ _ _ _ _ _) =
 simplifyDecl decl@(HSE.GDataInsDecl _ _ _ _ _ _) =
   notSupported "GADT style declarations" decl
 simplifyDecl decl@(HSE.ClassDecl _ _ _ _ _) = notSupported "Type classes" decl
-simplifyDecl decl@(HSE.InstDecl _ _ _ _) = notSupported "Type classes" decl
-simplifyDecl decl@(HSE.DerivDecl _ _ _ _) = notSupported "Type classes" decl
+simplifyDecl decl@(HSE.InstDecl _ _ _ _) = do
+  skipNotSupported "Instance declarations" decl
+  return ([], [], [])
```

Comment:
Would it make sense to move the `return ([],[],[])` into the definition of `skipNotSupported`?

Reply:
I don't think so. Not in every position, where `skipNotSupported` is used, is `([],[],[])` the result to return. One could of course modify `skipNotSupported` such that it takes `([],[],[])` as a third argument and returns it, but this would not make the code much shorter.
Additionally, I would say that the purpose of this function should be to only produce a warning and not to return some value as it is also used within larger `do` blocks like in line 144.

Reply:
Since the return type of `skipNotSupported` currently is `()`, all other usages of the function in a do-block just skip the result, which is still possible (and plausible) with the changed return value.